### PR TITLE
feat: added custom session replication prototype

### PR DESCRIPTION
**Pull request by mcollovati**
Adds custom components for session replication from kubernetes-scaling repo.
Autoconfiguration has been refactored to activate this feature only if spring-session is not present

Refrences #19

### CODE REVIEW

**Comment by mcollovati** [comment]
Demo application work with custom session replication on local cluster.
Tests still missing

File: azure-kit-demo/pom.xml [hunk 1]
```diff
@@ -55,6 +62,11 @@
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
                 <version>${spring.boot.version}</version>
+                <!--
```
Comment (MatthewVaadin) [inline azure-kit-demo/pom.xml:65]: Remove commented code

**Review by MatthewVaadin — CHANGES_REQUESTED** [review]

File: azure-kit-demo/pom.xml (hunk 1 above)
Comment (mcollovati) [reply azure-kit-demo/pom.xml]: done

**Review by MatthewVaadin — APPROVED** [review]

**Review by MatthewVaadin — CHANGES_REQUESTED** [review]
Add tests and JavaDoc for new classes

File: azure-kit-starter/src/main/java/com/vaadin/azure/starter/sessiontracker/backend/RedisConnector.java [hunk 2]
```diff
@@ -73,4 +71,9 @@ private byte[] getPendingKey(String clusterKey) {
         return BackendUtil.b("pending-" + clusterKey);
     }
 
+    @Autowired
+    void setRedisConnectionFactory(RedisConnectionFactory redisConnectionFactory) {
```
[the remark this answers is not in the document]
Comment (MatthewVaadin) [reply azure-kit-starter/src/main/java/com/vaadin/azure/starter/sessiontracker/backend/RedisConnector.java:75]: I'm not sure how to change these lines in `AzureKitConfiguration` if we do that:
```java
@Bean
@ConditionalOnBean(RedisConnectionFactory.class)
RedisConnector redisConnector() {
    return new RedisConnector();
}
```

[the remark this answers is not in the document]
Comment (mcollovati) [reply azure-kit-starter/src/main/java/com/vaadin/azure/starter/sessiontracker/backend/RedisConnector.java:75]: This should work
```
@Bean
@ConditionalOnBean(RedisConnectionFactory.class)
RedisConnector redisConnector(RedisConnectionFactory factory) {
    return new RedisConnector(factory);
}
```

**Review by MatthewVaadin — APPROVED** [review]

**Review by heruan — COMMENTED** [review]
There are still missing tests and JavaDoc, see https://github.com/vaadin/azure-kit/issues/28

**Pull request merged**